### ISSUE TITLE
Scheduled workflows should not run in the forks (limit to this repo)

### ISSUE THREAD

**Issue by sypets** [TYPO3/styleguide#388]
## Solution

While writing this, I found a solution (will create PR):

> In the workflow, you can use the property github.repository of [github context](https://help.github.com/en/actions/reference/contexts-and-expression-syntax-for-github-actions#github-context), or the [environment variable](https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables#default-environment-variables) GITHUB_REPOSITORY, to view the repository where the current workflow is running on.

```
 Jobs:
  build:
    if: (github.event_name == 'schedule' && github.repository == 'main_org/main_repo') || (github.event_name != 'schedule')
    . . .
```

https://github.com/orgs/community/discussions/26684

## Problem

**Is your question related to a problem? Please describe.**

TL;DR: Ideal would be to have the scheduled workflows, but not run them in the forks without having to turn them off individually.

-----

My problem is that the scheduled workflow runs also in the forks and it is annoying. You can disable it individually for your fork but AFAIK you have to disable it for all the workflows individually (so for "test", "tests-ac-mariadb-pdo-mysql", etc.), see https://docs.github.com/en/actions/managing-workflow-runs/disabling-and-enabling-a-workflow

I noticed it because the styleguide CI tests have started to fail and I am getting emails about it. I am also getting emails for another extension where I added a similar workflow (based on styleguide / brofix) and where my fork now daily runs the workflows.

(Apart from that it is unnecessary load because it makes no sense to constantly run the workflows in the (outdated) forks).

I am proposing a solution to generally turn off scheduled workflow runs for forks.

(The other solution is to delete the forks as soon as done. But you can't if you have a pending PR. A while ago I accidentally removed a fork with a pending PR. Unfortunately GitHub had no safeguard for that). 

I see a problem here because:

* styleguide is used as example in docs and other people may copy it (as I did) , later finding out this annoying side-effect. So this is not just a "styleguide" thing

I also realize the scheduled workflows are important, to be alerted early if things fail due to changes in core, for example.

**Comment by sbuerk** [comment]
Thanks for opening this issue. However, I'm here on the opposite side.

First of all, workflow in forks are only executed if accepted (once). And than it counts for all, until "manually" and seletive disabled again, like described by you.

But, what annoyes you is something I appreciate and need. GitHub do not send me a mail for failed scheuduled runs in the main repository, therefore I keep a fork which runs the scheduled tasks again so I get informed by email. 

Changing the workflows to disable scheduled runs outside the main repo will fully disable
this for all who want this. Re-emabling this is only possible with "commits" on branches and no simple sync possible etc and more extensive maintenance burden than disabling workflow if you do not want it.

So, personally don't like to change and narrow down the schedulred runs because it eleiminates the notification of failed schedule run for me.

**Comment by sypets** [comment]
@sbuerk ok, I understand. I assumed you were maintainer and would get the notifications for the main repo. 

From my end however, this is not so easy to resolve: For the workflows to run for my PRs, I have to enable them. But I don't really want them to run every day (indefinitely, since I rarely delete my forks and often reuse them for small PRs). Since, this is one extension, I can deal with it and change it. I would just not like other people to adopt this everywhere and scheduled workflows to run indefinitely in stale forks.

Personally, I have a real aversion against unnecessary energy consumption and use of CPU power, network bandwidth etc. I also got an objection on the line of use of resource power, energy and environment etc. when I tried to add the schedule workflow to another extension, so it's not just me.

I understand though that here and for you it is not unnecessary.

Personally I will use the suggested change for my extensions and contributions.
